gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopySource.java

```diff
@@ -382,9 +384,17 @@ public Void call() {
           setWorkUnitWatermark(workUnit, watermarkGenerator, copyEntity);
           computeAndSetWorkUnitGuid(workUnit);
           addLineageInfo(copyEntity, workUnit);
-          if (copyEntity instanceof CopyableFile && DistcpFileSplitter.allowSplit(this.state, this.targetFs)) {
-            workUnitsForPartition.addAll(DistcpFileSplitter.splitFile((CopyableFile) copyEntity, workUnit, this.targetFs));
+          if (copyEntity instanceof CopyableFile) {
+            CopyableFile castedCopyEntity = (CopyableFile) copyEntity;
+            fileSize = castedCopyEntity.getFileStatus().getLen();
+            workUnit.setProp(ServiceConfigKeys.WORK_UNIT_BYTE_SIZE, fileSize);
+            if (DistcpFileSplitter.allowSplit(this.state, this.targetFs)) {
+              workUnitsForPartition.addAll(DistcpFileSplitter.splitFile((CopyableFile) copyEntity, workUnit, this.targetFs));
+            } else {
+              workUnitsForPartition.add(workUnit);
+            }
           } else {
+            workUnit.setProp(ServiceConfigKeys.WORK_UNIT_BYTE_SIZE, 0);
```

Comment:
Can you leave a comment for when this condition would apply? If I recall correctly it's for some state files/post publish states